### PR TITLE
[12.0][IMP] account_bank_statement_import_txt_xlsx: Skip initial rows

### PR DESCRIPTION
**Pull request by MateuGForgeFlow**
Adds a field that allows to introduce an integer. This integer will be the number of initial rows that will be skipped in the imported file.
cc @LoisRForgeFlow

### CODE REVIEW

File: account_bank_statement_import_txt_xlsx/models/account_bank_statement_import_sheet_parser.py [hunk 1]
```diff
@@ -56,6 +56,12 @@ def parse(self, mapping, data_file, filename):
             journal.code,
             path.basename(filename),
         )
+        if mapping.delete_rows > 0 and name.lower().endswith('.csv'):
```
Comment (alexey-pelykh) [inline account_bank_statement_import_txt_xlsx/models/account_bank_statement_import_sheet_parser.py]: What's wrong with skipping XLS rows?

Comment (LoisRForgeFlow) [reply account_bank_statement_import_txt_xlsx/models/account_bank_statement_import_sheet_parser.py]: agree, it is better to skip if possible. Was there any impediment to skip?

File: account_bank_statement_import_txt_xlsx/models/account_bank_statement_import_sheet_mapping.py [hunk 2]
```diff
@@ -12,6 +12,13 @@ class AccountBankStatementImportSheetMapping(models.Model):
     name = fields.Char(
         required=True,
     )
+
+    delete_rows = fields.Integer(
```
Comment (LoisRForgeFlow) [inline account_bank_statement_import_txt_xlsx/models/account_bank_statement_import_sheet_mapping.py]: actually, I'm seeing now it is only a naming issue...
image

Can you change this field no somthing more accurate: `skipped_rows` or similar

Comment (MateuGForgeFlow) [reply account_bank_statement_import_txt_xlsx/models/account_bank_statement_import_sheet_mapping.py]: Done! Thank you @alexey-pelykh and @LoisRForgeFlow.

**Review by LoisRForgeFlow — APPROVED** [review]
LGTM

File: account_bank_statement_import_txt_xlsx/models/account_bank_statement_import_sheet_mapping.py [hunk 3]
```diff
@@ -12,6 +12,13 @@ class AccountBankStatementImportSheetMapping(models.Model):
     name = fields.Char(
         required=True,
     )
+
+    skipped_rows = fields.Integer(
+        string="Ignore First Rows",
+        help="The import will ignore the specified initial rows amount",
```
Comment (alexey-pelykh) [inline account_bank_statement_import_txt_xlsx/models/account_bank_statement_import_sheet_mapping.py]: > The number of lines to skip at the beginning of the file.

File: account_bank_statement_import_txt_xlsx/models/account_bank_statement_import_sheet_mapping.py [hunk 4]
```diff
@@ -12,6 +12,13 @@ class AccountBankStatementImportSheetMapping(models.Model):
     name = fields.Char(
         required=True,
     )
+
+    skipped_rows = fields.Integer(
```
Comment (alexey-pelykh) [inline account_bank_statement_import_txt_xlsx/models/account_bank_statement_import_sheet_mapping.py:16]: `rows_to_skip_start`? What if someone would add _end later

File: account_bank_statement_import_txt_xlsx/models/account_bank_statement_import_sheet_parser.py [hunk 5]
```diff
@@ -56,6 +56,12 @@ def parse(self, mapping, data_file, filename):
             journal.code,
             path.basename(filename),
         )
+        if mapping.skipped_rows > 0 and name.lower().endswith('.csv'):
```
Comment (alexey-pelykh) [inline account_bank_statement_import_txt_xlsx/models/account_bank_statement_import_sheet_parser.py:59]: If you skip rows in `_parse_lines` why this block is needed?

**Review by alexey-pelykh — CHANGES_REQUESTED** [review]
And travis fails

File: account_bank_statement_import_txt_xlsx/models/account_bank_statement_import_sheet_parser.py (hunk 5 above)
Comment (MateuGForgeFlow) [reply account_bank_statement_import_txt_xlsx/models/account_bank_statement_import_sheet_parser.py:59]: Hi @alexey-pelykh, in _parse_lines() the xls lines are skipped, here is the code for the .csv

**Comment by MateuGForgeFlow** [comment]
Hi @alexey-pelykh! Travis Fixed, thank you for the review!

File: account_bank_statement_import_txt_xlsx/models/account_bank_statement_import_sheet_mapping.py (hunk 3 above)
Comment (MateuGForgeFlow) [reply account_bank_statement_import_txt_xlsx/models/account_bank_statement_import_sheet_mapping.py]: Done!

File: account_bank_statement_import_txt_xlsx/models/account_bank_statement_import_sheet_mapping.py (hunk 4 above)
Comment (chienandalu) [reply account_bank_statement_import_txt_xlsx/models/account_bank_statement_import_sheet_mapping.py:16]: I'd go for a more precise `header_offset`

File: account_bank_statement_import_txt_xlsx/models/account_bank_statement_import_sheet_parser.py (hunk 5 above)
Comment (chienandalu) [reply account_bank_statement_import_txt_xlsx/models/account_bank_statement_import_sheet_parser.py:59]: This is not the best way to detect a csv, which in the end is just a plain text file. Why don't you just skip the proper offset in this `else`: https://github.com/OCA/bank-statement-import/blob/2e6a37f91e21d0d81b0062f7726bd6f427271f31/account_bank_statement_import_txt_xlsx/models/account_bank_statement_import_sheet_parser.py#L123-L128

Something like:
```python
     for i in range(mapping.skipped_rows):
         next(csv_or_xlsx)
     header = [value.strip() for value in next(csv_or_xlsx)]
```

**Review by chienandalu — COMMENTED** [review]
This would be a cool feature

Some comments though (also some tests would be nice)

**Pull request closed without merging**